packages/common-ethereum/src/codegen/codegen-controller.ts

```diff
@@ -133,31 +147,28 @@ export function prepareSortedAssets(
   projectPath: string
 ): Record<string, string> {
   const sortedAssets: Record<string, string> = {};
-  datasources
-    .filter((d) => !!d?.assets && (isRuntimeDs(d) || isCustomDs(d) || validateCustomDsDs(d)))
-    .forEach((d) => {
-      const addAsset = (name: string, value: FileReference) => {
-        // should do if covert to absolute
-        // if value.file is not absolute, the
-        const filePath = path.join(projectPath, value.file);
-        if (!fs.existsSync(filePath)) {
-          throw new Error(`Error: Asset ${name}, file ${value.file} does not exist`);
-        }
-        // We use actual abi file name instead on name provided in assets
-        // This is aligning with files in './ethers-contracts'
-        sortedAssets[parseContractPath(filePath).name] = value.file;
-      };
-
-      if (d.assets instanceof Map) {
-        for (const [name, value] of d.assets.entries()) {
-          addAsset(name, value);
-        }
-      } else {
-        Object.entries(d.assets).map(([name, value]) => {
-          addAsset(name, value as FileReference);
-        });
-      }
-    });
+
+  const addAsset = (name: string, value: FileReference) => {
+    // should do if covert to absolute
+    // if value.file is not absolute, the
+    const filePath = path.join(projectPath, value.file);
+    if (!fs.existsSync(filePath)) {
+      throw new Error(`Error: Asset ${name}, file ${value.file} does not exist`);
+    }
+    // We use actual abi file name instead on name provided in assets
+    // This is aligning with files in './ethers-contracts'
+    sortedAssets[parseContractPath(filePath).name] = value.file;
+  };
+
+  for (const d of datasources) {
+    if (!d.assets) continue;
+    if (!isRuntimeDs(d) && !isCustomDs(d) && !validateCustomDsDs(d)) continue;
+
+    for (const [name, value] of d.assets.entries()) {
```

Comment:
This has dropped support for the type being an object.

Reply:
Previously, support for objects was likely implemented for compatibility purposes. The `assets` is type Map. And the function generateAbis has already ensured that all incorrect assets are converted to Map.